python-sdk/nuscenes/panoptic/panoptic_utils.py

```diff
@@ -0,0 +1,206 @@
+"""
+nuScenes-panoptic utils.
+Code written by Motional and the Robot Learning Lab, University of Freiburg.
+"""
+from typing import Any, Dict, Iterable, List, Tuple
+
+import numpy as np
+from nuscenes.lidarseg.lidarseg_utils import colormap_to_colors, create_lidarseg_legend
+from nuscenes.utils.color_map import get_colormap
+from nuscenes.utils.data_io import load_bin_file
+
+create_panoptic_legend = create_lidarseg_legend
+
+
+def stuff_start_cat_id() -> int:
+    return 24
+
+
+def stuff_cat_ids(num_categories: int) -> List[int]:
+    """
+    :param num_categories: total number of classes.
+    :return: List of stuff class indices.
+    """
+    return list(range(stuff_start_cat_id(), num_categories, 1))
+
+
+def generate_panoptic_colors(colormap: Dict[str, Iterable[int]],
+                             name2idx: Dict[str, int],
+                             max_instances: int = 32000) -> np.ndarray:
+    """
+    Create an array of RGB values from a colormap for stuff categories, and random colors for thing instances. Note
+    that the RGB values are normalized between 0 and 1, not 0 and 255.
+    :param colormap: A dictionary containing the mapping from class names to RGB values.
+    :param name2idx: A dictionary containing the mapping form class names to class index.
+    :param max_instances: maximal number of instances.
+    :return: An array of colors.
+    """
+    np.random.seed(0)
```

Comment:
Make the seed value a parameter?

Reply:
this one we need it to be always with the same seed, so can get consistent colors for the same instance/panoptic ID (one row per color corresponds to a instance ID)